crypto/fn/fn_lib.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdbool.h>
+#include <openssl/opensslconf.h>
+#include <openssl/crypto.h>
+#include "fn_local.h"
+
+#define SIZE_IS_BITS 0
+
+static OSSL_FN *ossl_fn_new_internal(int limbs, bool securely)
+{
+    /* Total size of the whole OSSL_FN, in bytes */
+    int totalsize = sizeof(OSSL_FN) + limbs * sizeof(OSSL_FN_ULONG);
+
+    /*
+     * If the size ends up being smaller than the bookkeeping struct,
+     * we know that the size calculation has wrapped around and is
+     * therefore invalid.  Also, we don't allow zero byte numbers.
+     */
+    if (totalsize <= (int)sizeof(OSSL_FN))
+        return NULL;
+
+    OSSL_FN *ret = NULL;
+
+    if (securely)
+        ret = OPENSSL_secure_zalloc(totalsize);
+    else
+        ret = OPENSSL_zalloc(totalsize);
+
+    if (ret != NULL) {
+        ret->dsize = limbs;
+        ret->is_dynamically_allocated = 1;
+        ret->is_securely_allocated = securely;
+    }
+    return ret;
+}
+
+static void ossl_fn_free_internal(OSSL_FN *f, bool clear)
+{
+    if (f == NULL)
+        return;
+
+    int limbssize = f->dsize * sizeof(OSSL_FN_ULONG);
+    int totalsize = limbssize + sizeof(OSSL_FN);
```

Comment:
Again - it makes more sense to me for `limbsize` and `totalsize` to be `size_t` because that's what we pass  to `OPENSSL_clear_free` and `OPENSSL_cleanse`